### ISSUE TITLE
Feature Request: Midi output

### ISSUE THREAD

**Issue by DMJC** [CorsixTH/CorsixTH#1236]
Hi, Please add an option to allow MIDI output to be redirected to an external device/Midi port. I have a Roland Sound Canvas 55 Hardware Midi Synthesizer and would love to play the game as it was originally intended to be heard. 
pmidi -l on Linux shows
 24:0     CH345                             CH345 MIDI 1
I can set Dosbox to output Midi to this port, and pmidi -p 24:0 midi.mid will play a midi file on the synthesizer.

**Comment by TheCycoONE** [comment]
We use SDL_Mixer to handle our MIDI. I believe SDL Mixer at the moment only supports Mac, Windows, and Haiku for native MIDI. Could you take this request upstream?

**Comment by TheCycoONE** [comment]
@DMJC do you have a ticket with SDL that we can track?

**Comment by lewri** [comment]
No response

**Comment by TheCycoONE** [comment]
Going to leave this open, there's enough information in the ticket for us to act without further input from DMJC if the feature becomes a priority for someone.

**Comment by lewri** [comment]
On speaking with @ryanedewurth this morning:
FluidSynth and Timidity do have options for different soundfonts for software based MIDI. However, a workaround solution would be to probably use an external MIDI Mapper. 

It has been done for Linux in the past (see http://cowlark.com/amidimap/index.html ) -- and likely still possible now.

----

As a side note from #2180 -- MIDI support in Windows is horrible and needs a 3rd party mapper also. Or VirtualMIDISynth is a good software-based alternative that CorsixTH picks up right away (but music volume settings in game get ignored).

**Comment by lewri** [comment]
Does this feature become a bit more convoluted now for Windows because we forced that platform over to use fluidsynth?

**Comment by TheCycoONE** [comment]
No worse than it was before, we just switched from timidity to fluidsynth by default as an SDL2_mixer backend.

**Comment by mikedebian** [comment]
So yeah, I am one of those guys with hardware midi, with a loaded custom soundfont on the soundcard. Its exposed through Alsa and I have no problem using it in a lot of other software. Is there a command line switch or a config file to let Corsix-th work together with the specified alsa output port?

**Comment by TheCycoONE** [comment]
No. We are currently completely dependent on sdl mixer for our midi support. Upstream report: https://github.com/libsdl-org/SDL_mixer/issues/635

Looking for someone to take this issue.

**Comment by TheCycoONE** [comment]
Upstream SDL_mixer is no longer supporting native MIDI, so I have taken this in a different direction and added our own MIDI player using RtMidi. Device selection is built in from the beginning.

**Comment by TheCycoONE** [comment]
@mikedebian If you'd like to try out the attached PR, it includes in game options for setting the alsa port.

**Comment by mikedebian** [comment]
I am so sorry for being late to the party, but I have arrived and I have been jamming out to hardware midi in CorsixTH for the better part of the last hour. It works as intended, with no bugs, thanks :)

If I may add a suggestion, sometimes the midi port changes, but the identifier stays the same, and on that high note I have set this command export ALSA_OUTPUT_PORTS="Emu10k1 WaveTable:0", which in my case works, so I do not have to specify the port every time I want to play a midi file.

I have seen in the sound settings under corsix that I have to specify the port, in my case for this boot it is set to port 21. The full line so it makes sense is:

Emu10k1 WaveTable:Emu10k1 Port 0 21:0

So, the 21:0 is the bit that might change at times. How will Corsix handle this if the specified ports gets changed?
